app/scripts/controllers/monitoring.js

```diff
@@ -0,0 +1,311 @@
+      // Sort the longest keyword first.
+      keywords.sort(function(a, b){
+        return b.length - a.length;
+      });
+
+      // Convert the keyword to a case-insensitive regular expression for the filter.
+      filterExpressions = _.map(keywords, function(keyword) {
+        return new RegExp(_.escapeRegExp(keyword), "i");
+      });
+    };
+
+    // Only filter by keyword on certain fields.
+    var filterFields = [
+      'metadata.name'
+    ];
+
+    var filterForKeyword = function(resources) {
+      var filteredResources = resources;
+      if (!filterExpressions.length) {
+        return filteredResources;
+      }
+
+      // Find events that match all keywords.
+      angular.forEach(filterExpressions, function(regex) {
+        var matchesKeyword = function(event) {
+          var i;
+          for (i = 0; i < filterFields.length; i++) {
+            var value = _.get(event, filterFields[i]);
+            if (value && regex.test(value)) {
+              return true;
+            }
+          }
+
+          return false;
+        };
+
+        filteredResources = _.filter(filteredResources, matchesKeyword);
+      });
+      return filteredResources;
+    };
+
+    var filterAllResourcesForKeyword = function() {
+      $scope.filteredPods = filterForKeyword(ageFilteredPods);
+      $scope.filteredDeployments = filterForKeyword(ageFilteredDeployments);
+      $scope.filteredBuilds = filterForKeyword(ageFilteredBuilds);
+    };
+
+    var setPodLogVars = function(pod) {
+      $scope.logOptions.pods[pod.metadata.name] = {
+        container: pod.spec.containers[0].name
+      };
+      $scope.logCanRun.pods[pod.metadata.name] = !(_.includes(['New', 'Pending', 'Unknown'], pod.status.phase));
+    };
+
+    var setDeploymentLogVars = function(deployment) {   
+      $scope.logOptions.deployments[deployment.metadata.name] = {
+        container: $filter("annotation")(deployment, "pod")
+      };
+      var deploymentVersion = $filter("annotation")(deployment, "deploymentVersion");
+      if (deploymentVersion) {
+        $scope.logOptions.deployments[deployment.metadata.name].version = deploymentVersion;
+      }        
+      $scope.logCanRun.deployments[deployment.metadata.name] = !(_.includes(['New', 'Pending'], $filter('deploymentStatus')(deployment)));
+    };
+
+    var setBuildLogVars = function(build) {
+      $scope.logOptions.builds[build.metadata.name] = {
+        container: $filter("annotation")(build, "buildPod")
+      };
+      $scope.logCanRun.builds[build.metadata.name] = !(_.includes(['New', 'Pending', 'Error'], build.status.phase));
+    };
+
+    var filterPods = function() {
+      ageFilteredPods = _.filter($scope.pods, function(pod) {
+        if (!$scope.filters.hideOlderResources) {
+          return true;
+        }
+        if (pod.status.phase !== 'Succeeded' && pod.status.phase !== 'Failed') {
+          return true;
+        }
+        return false;
```

Comment:
Maybe just

```js
return pod.status.phase !== 'Succeeded' && pod.status.phase !== 'Failed';
```

Might be worth adding a filter since we have one for builds and deployments.